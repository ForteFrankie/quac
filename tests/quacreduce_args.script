#!/bin/bash

# Test that quacreduce argument parsing works correctly.
#
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR

set +e
exec 1>&2  # send echo to stderr to make output more readable
#set -x     # safe because we have no pipelines in this script


echo '*** Check input files for unique names'
x quacreduce --map cat --reduce cat job foo/bar.txt baz/qux.txt
echo $?
x quacreduce --map cat --reduce cat job foo/bar.txt baz/bar.txt
echo $?

echo '*** Fail if jobdir exists'
x mkdir jobexists
x quacreduce --map cat --reduce cat jobexists foo/bar.txt
echo $?

echo '*** Check that either --python xor --map and --reduce are specified'

echo '*** these should succeed'
#x quacreduce --python foo job1 input.txt
#echo $?
x quacreduce --map foo --reduce bar job2 input.txt
echo $?

echo '*** these should fail'
x quacreduce job input.txt
echo $?
x quacreduce --map bar job input.txt
echo $?
x quacreduce --reduce bar job input.txt
echo $?
x quacreduce --python foo --map bar job input.txt
echo $?
x quacreduce --python foo --reduce bar job input.txt
echo $?
x quacreduce --python foo --map bar --reduce baz job input.txt
echo $?

# clean up for subsequent tests
rm -Rf job job2

# exit with success
exit 0
