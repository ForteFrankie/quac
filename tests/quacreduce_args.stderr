*** Check input files for unique names
$ quacreduce --map cat --reduce cat job foo/bar.txt baz/qux.txt
0
$ quacreduce --map cat --reduce cat job foo/bar.txt baz/bar.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: input file basenames must be unique
2
*** Fail if jobdir exists
$ mkdir jobexists
$ quacreduce --map cat --reduce cat jobexists foo/bar.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: job dir "jobexists" already exists
2
*** Check that either --python xor --map and --reduce are specified
*** these should succeed
$ quacreduce --map foo --reduce bar job2 input.txt
0
*** these should fail
$ quacreduce job input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: --python or --map and --reduce must be specified
2
$ quacreduce --map bar job input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: --map and --reduce must both be specified if one is
2
$ quacreduce --reduce bar job input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: --map and --reduce must both be specified if one is
2
$ quacreduce --python foo --map bar job input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: --python cannot be specified with --map and/or --reduce
2
$ quacreduce --python foo --reduce bar job input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: --python cannot be specified with --map and/or --reduce
2
$ quacreduce --python foo --map bar --reduce baz job input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--partitions N]
                  [--sortdir DIR] [--sortmem N] [--update] [-h] [--unittest]
                  [--verbose]
                  JOBDIR FILE [FILE ...]
quacreduce: error: --python cannot be specified with --map and/or --reduce
2
